### PR TITLE
[MINOR][Documentation] Fix some minor descriptions in functions consistently with expressions

### PR DESCRIPTION
**Pull request by HyukjinKwon**
## What changes were proposed in this pull request?

This PR proposes to improve documentation and fix some descriptions equivalent to several minor fixes identified in https://github.com/apache/spark/pull/15677

Also, this suggests to change `Note:` and `NOTE:` to `.. note::` consistently with the others which marks up pretty.

## How was this patch tested?

Jenkins tests and manually. 

For PySpark, `Note:` and `NOTE:` to `.. note::` make the document as below:

**From**

**To**

### CODE REVIEW

**Comment by HyukjinKwon** [comment]
It seems something wrongly gone in my local so I can't run cran-check in my local. Please let me run this via Jenkins.

**Comment by HyukjinKwon** [comment]
Please let me cc @felixcheung and @srowen though if you don't mind.

**Comment by HyukjinKwon** [comment]
Oh, actually, it seems I should fix the ones in `functions.scala` too.

File: sql/core/src/main/scala/org/apache/spark/sql/functions.scala [hunk 1]
```diff
@@ -1145,15 +1147,16 @@ object functions {
...
+   * Generate a column with independent and identically distributed (i.i.d.) samples from
+   * the standard normal distribution.
    *
    * @group normal_funcs
    * @since 1.4.0
    */
   def randn(): Column = randn(Utils.random.nextLong)
 
   /**
-   * Partition ID of the Spark task.
```
Comment (HyukjinKwon) [inline sql/core/src/main/scala/org/apache/spark/sql/functions.scala]: This one was missed in https://github.com/apache/spark/commit/02f203107b8eda1f1576e36c4f12b0e3bc5e910e and https://github.com/apache/spark/commit/a08463b1d32348a81d0f148dfaf22741d5c23b1a

**Review by srowen — APPROVED** [review]

**Comment by HyukjinKwon** [comment]
It seems the newly added test seems too flaky. the specific test seems often failed. `randomized aggregation test - [typed, with partial + safe] - with grouping keys - with non-empty input` cc @liancheng

**Comment by HyukjinKwon** [comment]
retest this please

**Comment by felixcheung** [comment]
changes LGTM. thanks.
to nit pick, I find this "(Signed) shift the given value numBits right." confusing. Perhaps we should clarify this across all doc?

**Comment by srowen** [comment]
How would you write it? we do need to specify that it's a signed shift.

**Comment by felixcheung** [comment]
hmm, like "shift the signed value numBits to the right"?

**Comment by HyukjinKwon** [comment]
Ah, IMHO, it seems both are fine but actually it seems there is an unsigned shift one which is described as..

```
Unsigned shift the given value numBits right.
```

So.. if any of you does not feel strongly which one is better, I hope leave as it is because it seems at least consistent.

```
(Signed) shift the given value numBits right.
```

**Comment by felixcheung** [comment]
I think both sound odd, but it's a nit

**Comment by srowen** [comment]
It's not a shift of a signed value, but a 'signed' shift, meaning one that preserves the sign bit. I recognize the terms "unsigned shift" and "signed shift", FWIW.

**Comment by felixcheung** [comment]
Sure then.
Merged to master/branch-2.1

**Comment by HyukjinKwon** [comment]
Thank you both!

**Pull request closed without merging**